registration/api/v2/serializers.py

```diff
@@ -164,12 +165,47 @@ class Meta:
         }
 
 
+# pylint: disable=abstract-method
+class ThemeSchemaSerializerGenerator(serializers.Serializer):
+    """
+    Custom automatically generated serializer from the theme schemas.
+
+    This is needed to make the Swagger code generator correctly generate the
+    schema model in the frontend without requiring to duplicate the schema
+    there.
+    """
+    def __init__(self, *args, **kwargs):
+        """
+        Start with empty serializer and add fields from both theme schemas
+        """
+        super().__init__(*args, **kwargs)
+
+        theme_schema_combined = {
+            **theme_schema_v0['properties'],
+            **theme_schema_v1['properties']
+        }
+        for key, value in theme_schema_combined.items():
+            field_type = None
+            if key == 'version':
+                field_type = serializers.IntegerField()
+            elif value == ref('flag'):
+                field_type = serializers.BooleanField(required=False)
+            else:
+                field_type = serializers.CharField(
+                    max_length=7,
+                    required=False
+                    # TODO: Add a color validator here
+                )
+            self.fields[key] = field_type
+
+
 class OpenEdXInstanceConfigSerializer(serializers.ModelSerializer):
     """
     Serializer with configuration details about the user's Open edX instance.
     """
 
     user = serializers.HiddenField(default=serializers.CurrentUserDefault())
+    draft_theme_config = ThemeSchemaSerializerGenerator(required=False)
```

Comment:
The `draft_theme_config` is where we store the theming variables before they are actually deployed using the `Deploy` button in the new console. This is just a temporary storage for changes made by the user.

Reply:
Thanks for the explanation. Given that rationale, shouldn't there be two different theme configs (the draft one and the effective one, currently used in the instance) ? If we won't keep the effective theme config in the db, is there a reason to name this as "draft"? If this "draft" theme config is kept in the db after the theme is deployed, it will no longer be a draft and the name will become misleading.

Reply:
@alanoe, both the draft and the final theme configuration values are stored in the database. Just that the latter will have been deployed on an appserver - i.e., the user can keep changing the theme configuration from the frontend and each change will get stored and overwrite the already stored draft configuration in the database. But when the user clicks a button to deploy the changes, the version of draft configuration is copied to the final configuration and the instance is redeployed to apply the changes.